### PR TITLE
Add request validation in work order handler.

### PR DESCRIPTION
**Pull request by Ram-srini**
1. Remove request validation from work order jrpc class
2. Add request request validation in  work order handler
3. Remove empty optional field from work order request
3. Fix load worker if proof data is empty

Signed-off-by: Ramakrishna Srinivasamurthy <ramakrishna.srinivasamurthy@intel.com>

### CODE REVIEW

File: tests/json_requests/json_03.json [hunk 1]
```diff
@@ -11,22 +11,19 @@
...
         "dataEncryptionAlgorithm": "AES-GCM-256",
         "encryptedSessionKey": "sessionkey",
         "sessionKeyIv": "Iv",
         "requesterNonce": "",
         "encryptedRequestHash": "requesthash",
         "requesterSignature": "",
         "inData": [
-		   {"index": 2,
-                    "dataHash": "",
+		   {        "index": 2,
```
Comment (manojgop) [inline tests/json_requests/json_03.json]: formatting

File: examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py [hunk 2]
```diff
@@ -96,8 +100,12 @@ def WorkOrderGetResult(self, **params):
             - jrpc_id is JRPC id of response
             - response is the response object to be returned
         """
-
         wo_id = params["workOrderId"]
+        if not is_hex(wo_id):
+            logging.error("Invalid work order Id")
+            return create_jrpc_response(id, JsonRpcErrorCode.INVALID_PARAMETER,
+                "Invalid work order Id")
```
[the remark this answers is not in the document]
Comment (Ram-srini) [reply examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py]: Changed to JSONRPCDispatchException

File: examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py [hunk 3]
```diff
@@ -140,6 +148,31 @@ def WorkOrderSubmit(self, **params):
 
         wo_id = params["workOrderId"]
         input_json_str = params["raw"]
+        input_value_json = json.loads(input_json_str)
+        jrpc_id = input_value_json["id"]
+
+        req_validator = WorkOrderRequestValidator()
+        valid, err = req_validator.validate_parameters(input_value_json["params"])
```
[the remark this answers is not in the document]
Comment (Ram-srini) [reply examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py]: Done

File: tests/json_requests/json_03.json (hunk 1 above)
Comment (Ram-srini) [reply tests/json_requests/json_03.json]: formatted

File: examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py [hunk 4]
```diff
@@ -140,6 +149,37 @@ def WorkOrderSubmit(self, **params):
 
         wo_id = params["workOrderId"]
         input_json_str = params["raw"]
+        input_value_json = json.loads(input_json_str)
+        jrpc_id = input_value_json["id"]
```
Comment (danintel) [inline examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py]: LGTM flags this as never used:

`jrpc_id = input_value_json["id"]`
--
  **| NEW The value assigned to local variable 'jrpc_id' is never used.**

Comment (Ram-srini) [reply examples/common/python/connectors/direct/tcs_listener/tcs_work_order_handler.py]: Addressed it.

File: examples/common/python/utility/hex_utils.py [hunk 5]
```diff
@@ -26,7 +26,14 @@ def hex_to_utf(binary):
     return binascii.hexlify(binary).decode("UTF-8")
 
 def is_hex(s):
-    return all(c in string.hexdigits for c in s)
+    if len(s):
```
[the remark this answers is not in the document]
Comment (Ram-srini) [reply examples/common/python/utility/hex_utils.py]: Added

[the remark this answers is not in the document]
Comment (rranjan3) [reply examples/common/python/utility/hex_utils.py]:
```
def is_hex(s):
    try:
        int(s, 16)
        return True
    except ValueError:
        return False
```
This implementation should suffice and would be more efficient in most cases.

File: examples/common/python/utility/hex_utils.py [hunk 6]
```diff
@@ -25,8 +25,18 @@ def pretty_ids(ids):
 def hex_to_utf(binary):
```
Comment (danintel) [inline examples/common/python/utility/hex_utils.py]: LGTM says there is a unused import on line 16:
```

import string
--
  **| NEW Import of 'string' is not used.**
```

(for details, see https://lgtm.com/projects/g/hyperledger-labs/trusted-compute-framework/rev/pr-42871d48af4833557e75cb15523d85dbcdd222cc )

File: examples/common/python/utility/hex_utils.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (danintel) [reply examples/common/python/utility/hex_utils.py]: I do not see where you added this comment. The code no longer allows a `0x` prefix. Have you decided not to allow a `0x` prefix?

File: examples/common/python/utility/hex_utils.py (hunk 6 above)
Comment (Ram-srini) [reply examples/common/python/utility/hex_utils.py]: Removed unused import

File: examples/common/python/utility/hex_utils.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Ram-srini) [reply examples/common/python/utility/hex_utils.py]: I replaced the implementation with int converter that will take care of "0X" or "0x" prefix in hex string.

**Pull request merged**